File: sdk/python/sdk/zrok/zrok/overview.py
from zrok.environment.root import Root
import urllib3


def Overview(root: Root) -> str:
    if not root.IsEnabled():
        raise Exception("environment is not enabled; enable with 'zrok enable' first!")

    http = urllib3.PoolManager()
    apiEndpoint = root.ApiEndpoint().endpoint
    try:
        response = http.request(
            'GET',
            apiEndpoint + "/api/v1/overview",
            headers={
                "X-TOKEN": root.env.Token
            })
    except Exception as e:
        raise Exception("unable to get account overview", e)
    return response.data.decode('utf-8')

File: sdk/golang/sdk/overview.go
package sdk

import (
	"errors"
	"fmt"
	"github.com/openziti/zrok/environment/env_core"
	"io"
	"net/http"
)

func Overview(root env_core.Root) (string, error) {
	if !root.IsEnabled() {
		return "", errors.New("environment is not enabled; enable with 'zrok enable' first!")
	}

	client := &http.Client{}
	apiEndpoint, _ := root.ApiEndpoint()
	req, err := http.NewRequest("GET", fmt.Sprintf("%v/api/v1/overview", apiEndpoint), nil)
	if err != nil {
		return "", err
	}
	req.Header.Add("X-TOKEN", root.Environment().AccountToken)
	resp, err := client.Do(req)
	if err != nil {
		return "", err
	}

	json, err := io.ReadAll(resp.Body)
	if err != nil {
		return "", err
	}
	_ = resp.Body.Close()

	return string(json), nil
}
